qcengine/programs/entos.py

```diff
@@ -0,0 +1,220 @@
+"""
+Calls the entos executable.
+"""
+
+from typing import Any, Dict, Optional, List
+
+from qcelemental.models import Result, FailedOperation
+from ..util import execute, popen
+from qcelemental.util import which, safe_version, parse_version, which_import
+from ..exceptions import UnknownError
+
+from .model import ProgramHarness
+
+
+class EntosHarness(ProgramHarness):
+    _defaults = {
+        "name": "entos",
+        "scratch": True,
+        "thread_safe": False,
+        "thread_parallel": True,
+        "node_parallel": False,
+        "managed_memory": True,
+    }
+    version_cache: Dict[str, str] = {}
+
+    class Config(ProgramHarness.Config):
+        pass
+
+    def __init__(self, **kwargs):
+        super().__init__(**{**self._defaults, **kwargs})
+
+    def found(self, raise_error: bool = False) -> bool:
+        return which('entos', return_bool=True, raise_error=raise_error, raise_msg='Please install via XXX')
+
+    def get_version(self) -> str:
+        self.found(raise_error=True)
+
+        which_prog = which('entos')
+        if which_prog not in self.version_cache:
+            with popen([which_prog, '--version']) as exc:
+                exc["proc"].wait(timeout=15)
+            self.version_cache[which_prog] = safe_version(exc["stdout"].split()[2])
+
+        return self.version_cache[which_prog]
+
+    def compute(self, input_data: 'ResultInput', config: 'JobConfig') -> 'Result':
+        """
+        Run entos
+        """
+        # Check if entos executable is found
+        self.found(raise_error=True)
+
+        # Check entos version
+        if parse_version(self.get_version()) < parse_version("0.5"):
+            raise TypeError("entos version '{}' not supported".format(self.get_version()))
+
+        # Setup the job
+        job_inputs = self.build_input(input_data, config)
+
+        # Run entos
+        proc = self.execute(job_inputs)
+
+        if isinstance(proc, FailedOperation):
+            return proc
+        else:
+            # If execution succeeded, collect results
+            result = self.parse_output(proc["outfiles"], input_data)
+            return result
+
+    def execute(self, inputs, extra_outfiles=None, extra_commands=None, scratch_name=None, timeout=None):
```

Comment:
Can you add typing info here from the base harness?

Reply:
There is no typing info currently for the execute function from the base harness.